File: Python/LeetCode/sortArray.py
class Solution:
    @staticmethod
    def sortArray(nums: List[int]) -> List[int]:
        def merge(nums: List[int]) -> List[int]:
            if len(nums) == 1:
                return nums

            middle = len(nums) // 2
            left = merge(nums[:middle])
            right = merge(nums[middle:])

            return mergeTwoArray(left, right)

        def mergeTwoArray(a: List[int], b: List[int]) -> List[int]:
            result = []
            i = j = 0

            while i < len(a) and j < len(b):
                if a[i] < b[j]:
                    result.append(a[i])
                    i += 1
                else:
                    result.append(b[j])
                    j += 1

            if i < len(a):
                result += a[i:]
            if j < len(b):
                result += b[j:]

            return result
        
        return merge(nums)

File: CSharp/LeetCode/SortArray.cs
public class Solution {
    public int[] SortArray(int[] nums) 
    {
        if (nums.Length <= 1)
            return nums;

        int middle = nums.Length / 2;
        int[] left = SortArray(nums[..middle]);
        int[] right = SortArray(nums[middle..]);

        return MergeTwoArray(left, right);
    }

    private int[] MergeTwoArray(int[] a, int[] b)
    {
        var result = new List<int>();
        int i = 0;
        int j = 0;

        while (i < a.Length && j < b.Length)
        {
            if (a[i] < b[j])
                result.Add(a[i++]);
            else
                result.Add(b[j++]);
        }

        if (i < a.Length)
            result.AddRange(a[i..]);
        if (j < b.Length)
            result.AddRange(b[j..]);

        return result.ToArray();
    }
}
